### ISSUE TITLE
Many compilation warnings

### ISSUE THREAD

**Issue by xDGameStudios** [godotengine/godot#28649]
**Godot version:**
Latest master #7b64a24

**OS version:**
MacOSX Mojave
Scons v3.0.1

**Issue description:**
there are many compilation warnings:
```
thirdparty/bullet/LinearMath/btVector3.h:335:7: warning: argument value 10880 is outside the valid range [0, 255] [-Wargument-outside-range]
                y = bt_splat_ps(y, 0x80);
                    ^~~~~~~~~~~~~~~~~~~~
thirdparty/bullet/LinearMath/btVector3.h:43:29: note: expanded from macro 'bt_splat_ps'
#define bt_splat_ps(_a, _i) bt_pshufd_ps((_a), BT_SHUFFLE(_i, _i, _i, _i))
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
thirdparty/bullet/LinearMath/btVector3.h:41:33: note: expanded from macro 'bt_pshufd_ps'
#define bt_pshufd_ps(_a, _mask) _mm_shuffle_ps((_a), (_a), (_mask))
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Steps to reproduce:**
Try compile source

**Comment by akien-mga** [comment]
I can't reproduce that, please give more details on your OS, compiler, exact version you're building, exact build instruction.

**Comment by akien-mga** [comment]
And please run that build instruction again with `verbose=yes`, and post the full build output.

**Comment by johngirvin** [comment]
I get the same error and can't build macos or ios with arch x86/x64 after upgrading to Mojave / Xcode 11. I'd built these before on High Sierra without issue. iphone arm builds are fine.

**Godot Version:**
commit 66baa3b633fe904ea0d90a9688d602d9f3a0b3bd (HEAD, tag: 3.1.1-stable)

**Build Commands:**
scons verbose=yes platform=iphone target=debug arch=x86 tools=no
scons verbose=yes platform=iphone target=debug arch=x86_64 tools=no
scons verbose=yes platform=osx target=debug arch=x86 tools=no
scons verbose=yes platform=osx target=debug arch=x86_64 tools=no
(FWIW target & tools settings seem to be irrelevant to the problem)

**Build Logs:**
[scons.iphone.debug.x64.log](https://github.com/godotengine/godot/files/3685215/scons.iphone.debug.x64.log)
[scons.iphone.debug.x86.log](https://github.com/godotengine/godot/files/3685216/scons.iphone.debug.x86.log)
[scons.osx.debug.x64.log](https://github.com/godotengine/godot/files/3685255/scons.osx.debug.x64.log)
[scons.osx.debug.x86.log](https://github.com/godotengine/godot/files/3685265/scons.osx.debug.x86.log)

**Comment by akien-mga** [comment]
Duplicate of #29503 and #32301, fixed already in the `master`, `3.1` and `3.0` branches (#32303).
